optimum/pipelines/diffusers/pipeline_stable_diffusion.py

```diff
@@ -210,25 +349,36 @@ def __call__(
             if callback is not None and i % callback_steps == 0:
                 callback(i, t, latents)
 
-        latents = 1 / 0.18215 * latents
-        # image = self.vae_decoder(latent_sample=latents)[0]
-        # it seems likes there is a strange result for using half-precision vae decoder if batchsize>1
-        image = np.concatenate(
-            [self.vae_decoder(latent_sample=latents[i : i + 1])[0] for i in range(latents.shape[0])]
-        )
+        if output_type == "latent":
+            image = latents
+            has_nsfw_concept = None
+        else:
+            latents = 1 / self.vae_decoder.config.get("scaling_factor", 0.18215) * latents
+            # it seems likes there is a strange result for using half-precision vae decoder if batchsize>1
```

Comment:
Interesting! Not sure if it's related to this: https://huggingface.slack.com/archives/C03UQJENJTV/p1688029497835759